cpp/src/io/parquet/page_enc.cu

```diff
@@ -1337,11 +1343,206 @@ static __device__ void byte_reverse128(__int128_t v, void* dst)
                d_char_ptr);
 }
 
-__device__ void get_extremum(const statistics_val* stats_val,
-                             statistics_dtype dtype,
-                             void* scratch,
-                             const void** val,
-                             uint32_t* len)
+/**
+ * @brief Test to see if an array contains a valid UTF-8 string.
+ *
+ * @param str device_span containing array to test.
+ * @return true if the array contains a valid UTF-8 string.
+ */
+static __device__ bool is_valid_utf8_string(device_span<uint8_t const> str)
```

Comment:
Don't need to add `static` keyword if the function is not part of a class, or if it is not to be called outside of its namespace.

Reply:
The function is not meant to be visible outside the compilation unit, so I made it static.  Is this not necessary?

Reply:
Might we consider an unnamed namespace instead?

Reply:
This new C++ amazes me :)  Finding it hard to shake my dinosaur C ways.

Reply:
Tell me about it. It feels like a completely different language to the fancy new C99. :)